src/v/kafka/server/group.cc

```diff
@@ -2288,6 +2291,14 @@ group::offset_commit_stages group::store_offsets(offset_commit_request&& r) {
     std::vector<std::pair<model::topic_partition, offset_metadata>>
       offset_commits;
 
+    const auto expiry_timestamp = [&r]() -> std::optional<model::timestamp> {
```

Comment:
good point. seems there may be some assumption about "reasonable" clock sync in kafka (and in parts of redpanda). since retention times are more on the order of like days, that's more than enough provided by ntp.

>serialized the duration (instead of timestamp) so that we always overestimate the retention time.

so in this case the base time is effectively always reset? something like `duration + <time point = when data was recover>` or something like this?

Reply:
>  since retention times are more on the order of like days,

Ah okay, I thought its much smaller, like seconds. Then the current approach is good I think.

> so in this case the base time is effectively always reset? something like duration + <time point = when data was recover> or something like this?

yes, exactly.